aQute.libg/src/aQute/libg/uri/URIUtil.java

```diff
@@ -41,4 +44,83 @@ public static URI resolve(URI baseURI, String reference) throws URISyntaxExcepti
 		return baseURI.resolve(refURI);
 	}
 
+	private static final URI EMPTYURI = URI.create("");
+
+	/**
+	 * Attempts to fetch a path on the file system for the given URI string.
+	 * Tries a few more tricks than the standard method of
+	 * <tt>Paths.get(new URI(uriString))</tt> - it can handle plain paths, and
+	 * also nested URI pseudo-schemes like <tt>reference:</tt> and
+	 * <tt>jar:</tt>. Examples:
+	 * <ul>
+	 * <li>/path/to/file => /path/to/file</li>
+	 * <li>reference:file:/path/to/file => /path/to/file</li>
+	 * <li>jar:file:/path/to/file.jar!/some/contained/element =>
+	 * /path/to/file.jar</li>
+	 * <li>http://server/path/to/file => <tt>null</tt></li>
+	 * </ul>
+	 *
+	 * @param uriString The string containing the URI for which we are
+	 *            attempting to construct a path.
+	 * @return The method's best guess as to which file on the local filesystem
+	 *         this URI refers, or <tt>null</tt> if it's invalid or not a local
+	 *         filesystem URI.
+	 */
+	public static Path fromURI(String uriString) {
+		if (uriString == null || uriString.isEmpty()) {
+			return null;
+		}
+
+		try {
+			URI uri = resolve(EMPTYURI, uriString);
+			while (uriString != null && !uriString.isEmpty()) {
+
+				final String scheme = uri.getScheme();
+				if (scheme == null) {
+					try {
+						return Paths.get(uriString);
+					} catch (InvalidPathException e) {
+						return null;
+					}
+				}
+
+				switch (uri.getScheme()
+					.toLowerCase()) {
+					case "file" :
+						return Paths.get(uri);
+					case "jar" :
+					case "bundle" :
+					case "zip" :
+						uriString = uri.getSchemeSpecificPart();
+						int index = uriString.indexOf('!');
+						if (index >= 0) {
+							uriString = uriString.substring(0, index);
+						}
+						if (uriString == null) {
+							return null;
+						}
+						uri = resolve(EMPTYURI, uriString);
+						break;
+					case "reference" :
+						uriString = uri.getSchemeSpecificPart();
+						uri = resolve(EMPTYURI, uriString);
+						break;
+					default :
+						uriString = uri.getSchemeSpecificPart();
+						try {
+							uri = new URI(uriString);
+							if (uri.getScheme() == null) {
+								return null;
+							}
+						} catch (URISyntaxException e) {
```

Comment:
Is this additional try/catch for URISytaxException needed since the code is already in a try/catch for URISytaxException which will return null?

Reply:
No, it is not.